src/WebJobs.Script.WebHost/Standby/WarmUpConstants.cs

```diff
@@ -9,5 +9,6 @@ public static class WarmUpConstants
     {
         public const string FunctionName = "WarmUp";
         public const string AlternateRoute = "CSharpHttpWarmup";
+        public const string WarmupTrigger = "WarmUpTrigger";
```

Comment:
Likely this trigger won't include any custom programming model or inputs. If we're just looking for a trigger type that allows string/JObject to be passed, we could either promote ManualTrigger outside of the host (since it's already a concept), or perhaps have the warmup function be an HttpTrigger? This way we don't have to define a new public binding type.

Reply:
@mathewc, if the concern is for us to do the additional work, I made the PR for the warmup extension [here](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/590).

But if you have other thoughts, please let us know. We can discuss offline, and if we think leveraging our existing extensions is the right approach, I would be happy to update this PR.

Reply:
It's not about the work involved to add a new trigger type, it's about increasing concept count (a new binding) with no real benefit. If we just move ManualTrigger out into the Extensions library (peer to Timer and other core built ins) user's can use that. That also has the benefit of addressing the issue that currently precompiled functions cannot use ManualTrigger.

Reply:
Thanks Mathew! Yeah I think initially we didn't go with ManualTrigger as it's part of the host right now and as you said working with precompile would be a challenge.

I think your suggestion makes sense to move it to extensions repo. I am guessing, from the host side it should not be a breaking change that `ManualTrigger` with not be passed to sdk as `NoAutomaticTrigger` anymore (from my understanding that's how it works today).

I am ok with leveraging the Manual trigger. I will wait to hear @fabiocav remarks before making that change.

Reply:
As per offline discussion, we will be going ahead with a new `WarmupTrigger`.